src/ibek/ioc_factory.py

```diff
@@ -0,0 +1,225 @@
+
+            # add in SubEntities if any
+            for sub_entity in definition.sub_entities:
+                # find the Entity Class that the SubEntity represents
+                entity_cls = self._entity_classes[sub_entity.type]
+                # get the SubEntity arguments
+                sub_args_dict = sub_entity.model_dump()
+                # jinja render any references to parent Args in the SubEntity Args
+                for key, arg in sub_args_dict.items():
+                    sub_args_dict[key] = UTILS.render(entity, arg)
+                # cast the SubEntity to its concrete Entity subclass
+                cast_entity = entity_cls(**sub_args_dict)
+                # recursively scan the SubEntity for more SubEntities
+                scan_sub_entities(cast_entity)
+
+        for entity in ioc_instance.entities:
+            scan_sub_entities(entity)
+
+        ioc_instance.entities = all_entities
+
+    def _make_entity_model(
+        self, definition: Definition, support: Support
+    ) -> Type[Entity]:
+        """
+        Create an Entity Model from a Definition instance and a Support instance.
+        """
+
+        def add_arg(name, typ, description, default):
+            if default is None:
+                default = PydanticUndefined
+            args[name] = (
+                typ,
+                FieldInfo(description=description, default=default),
+            )
+
+        args: Dict[str, Tuple[type, Any]] = {}
+        validators: Dict[str, Any] = {}
+
+        # fully qualified name of the Entity class including support module
+        full_name = f"{support.module}.{definition.name}"
+
+        # add in each of the arguments as a Field in the Entity
+        for arg in definition.args:
+            full_arg_name = f"{full_name}.{arg.name}"
+            arg_type: Any
+
+            if isinstance(arg, ObjectArg):
+
+                @field_validator(arg.name, mode="after")
+                def lookup_instance(cls, id):
+                    return get_entity_by_id(id)
+
+                validators[full_arg_name] = lookup_instance
+                arg_type = object
+
+            elif isinstance(arg, IdArg):
+                arg_type = str
+
+            elif isinstance(arg, EnumArg):
+                # Pydantic uses the values of the Enum as the options in the schema.
+                # Here we arrange for the keys to be in the schema (what a user supplies)
+                # but the values to be what is rendered when jinja refers to the enum
+                enum_swapped = {}
+                for k, v in arg.values.items():
+                    enum_swapped[str(v) if v else str(k)] = k
+                val_enum = EnumVal(arg.name, enum_swapped)  # type: ignore
+                arg_type = val_enum
+
+            else:
+                # arg.type is str, int, float, etc.
+                arg_type = getattr(builtins, arg.type)
+            default = getattr(arg, "default", None)
```

Comment:
Args can have a default. If they don't have a default we want to pass None to add_arg. I think I'm missing the reason for the question though?

Reply:
Oh right, I think it just looked a bit confusing on first glance.

Could `default: Optional[Any] = None` be added to `Arg`, so that you don't need `getattr`? Is pydantic OK with that?

Reply:
All args already have their own default as e.g.
https://github.com/epics-containers/ibek/blob/a1631b681f1e18f8df175e4192cae192ed01226c/src/ibek/args.py#L41

Maybe this is a holdover from SchemaAPI and is now unecessary. I'll have to check once the major refactor is complete.

Reply:
I was able to simplify this down to 
```
            add_arg(arg.name, arg_type, arg.description, getattr(arg, "default"))
```
But not sure if that is what you were getting at?